core/src/main/scala/kafka/controller/ControllerContext.scala

```diff
@@ -100,7 +140,7 @@ class ControllerContext {
 
   def removeTopic(topic: String) = {
     partitionLeadershipInfo = partitionLeadershipInfo.filter { case (topicPartition, _) => topicPartition.topic != topic }
-    partitionReplicaAssignment = partitionReplicaAssignment.filter { case (topicPartition, _) => topicPartition.topic != topic }
+    partitionReplicaAssignmentUnderlying.remove(topic)
```

Comment:
Should we also update replicasOnOfflineDirs accordingly?

Reply:
I think we should update replicasOnOfflineDirs accordingly. However I don't want to simply iterate through the current replicasOnOfflineDirs to filter out the topic since doing that will be very slow in a large cluster.
Since clearing the replicasOnOfflineDirs is an orthogonal change, I'll think more about whether it's important and if so how to do it. I feel a separate PR is probably better. Comment?